### PR TITLE
Fix LineSegment2D.TryIntersect to return true only when segments inte…

### PR DESCRIPTION
**Pull request by imlex**
…rsect not lines.

### CODE REVIEW

File: src/Spatial.Tests/Euclidean/LineSegment2DTests.cs [hunk 1]
```diff
@@ -139,6 +139,19 @@ public void IntersectWithTest(string s1, string e1, string s2, string e2, string
             Assert.AreEqual(e, result);
         }
 
+        [TestCase("0,0", "-2,-2", "1,0", "1,2", "1,1")]
```
Comment (JohanLarsson) [inline src/Spatial.Tests/Euclidean/LineSegment2DTests.cs]: Why not add the test cases to the existing test?

File: src/Spatial.Tests/Euclidean/LineSegment2DTests.cs [hunk 2]
```diff
@@ -128,7 +128,7 @@ public void ClosestPointToWithinSegment(string start, string end, string point,
 
         [TestCase("0,0", "2,2", "1,0", "1,2", "1,1")]
         [TestCase("0,0", "2,2", "0,1", "2,1", "1,1")]
-        [TestCase("0,0", "2,2", "-1,-5", "-1,0", "-1,-1")]
+        [TestCase("-1,-1", "2,2", "-1,-5", "-1,0", "-1,-1")]
```
Comment (JohanLarsson) [inline src/Spatial.Tests/Euclidean/LineSegment2DTests.cs]: We should add a test for parallel segments but that does not have to be in this PR.

File: src/Spatial/Euclidean/LineSegment2D.cs [hunk 3]
```diff
@@ -157,9 +157,10 @@ public bool TryIntersect(LineSegment2D other, out Point2D intersection, Angle to
             var s = other.StartPoint.VectorTo(other.EndPoint);
 
             var t = (q - p).CrossProduct(s) / r.CrossProduct(s);
+            var u = (p - q).CrossProduct(r) / s.CrossProduct(r);
```
Comment (JohanLarsson) [inline src/Spatial/Euclidean/LineSegment2D.cs]: Why are both cross products needed?

File: src/Spatial.Tests/Euclidean/LineSegment2DTests.cs (hunk 1 above)
Comment (imlex) [reply src/Spatial.Tests/Euclidean/LineSegment2DTests.cs]: Why not? One is for positive result, another is for negative.

**Pull request merged**